distributor.php

```diff
@@ -55,14 +55,29 @@ function( $class ) {
 );
 
 /**
- * Require PHP version 5.6 - throw an error if the plugin is activated on an older version.
+ * Require PHP 7.4+, WP 5.7+ - throw an error if the plugin is activated on an older version.
  */
 register_activation_hook(
```

Comment:
These changes prevent activation of the plugin on unsupported versions of WP, PHP.

There is now the small risk a site will have an active plugin and update to a version it no longer supports. How have we handled that in the past?

Reply:
So I was trying to find a version of this in our plugins and couldn't quite remember where it's at but what we've done in the past is have some code in the main plugin file (so this file in this case) that checks if our requirements are met and if not, output an admin notice and stop execution of the plugin. Not our plugin but here's a similar approach in a Woo plugin: https://github.com/woocommerce/google-listings-and-ads/blob/develop/google-listings-and-ads.php#L41 and https://github.com/woocommerce/google-listings-and-ads/blob/develop/src/Internal/Requirements/VersionValidator.php#L22

Reply:
In 6187c5625d1f612a7715b3be0c9da9bf0c725c7a I've added a first pass at showing a notice if the plugin fails to meet minimum version requirements.

Honestly, it seems like a lot for something that ought to be quite minor but I am not sure if there is much that can be done about that. For this to work, the plugin's main file will always need to be compatible with earlier versions of WP/PHP. The up-to-date code will need to come later.

Should I move some of the bootstrapping code so `distributor.php` can focus on activation & a bootstrapper can focus on bootstrapping?